packages/ui/src/Table/utils.tsx

```diff
@@ -0,0 +1,183 @@
+  const rendererProps: HeaderProps<D> = {
+    rows: [],
+    data: [],
+  }
+
+  props.table.getRowModel().rows.forEach((row) => {
+    const item = prepareRow(row)
+
+    rendererProps.rows.push(item)
+    rendererProps.data.push(item.original)
+  })
+
+  return rendererProps
+}
+
+export const getColumnDef = <D extends RowData>({
+  id,
+  width: size,
+  minWidth: minSize,
+  maxWidth: maxSize,
+  Header,
+  Footer,
+  Cell,
+  align,
+  filterable,
+  disableSortBy,
+  disableResizing,
+  canHide: enableHiding,
+  sortType: sortingFn,
+  accessor,
+}: ColumnType<D>): ColumnDef<D> => {
+  const commonDef: Omit<ColumnDef<D>, 'accessorFn' | 'accessorKey'> = {
+    id,
+    header: typeof Header === 'function' ? (props) => Header(getRendererProps(props)) : Header,
+    footer: typeof Footer === 'function' ? (props) => Footer(getRendererProps(props)) : Footer,
+    enableHiding,
+    sortDescFirst: false,
+    enableSorting: true,
+    meta: { align },
+    enableResizing: !disableResizing,
+  }
+
+  if (size != undefined) {
+    commonDef.size = size
+  }
+
+  if (minSize != undefined) {
+    commonDef.minSize = minSize
+  }
+
+  if (maxSize !== undefined) {
+    commonDef.maxSize = maxSize
+  }
+
+  if (sortingFn != undefined) {
+    commonDef.sortingFn = (rowA, rowB) => sortingFn(prepareRow(rowA), prepareRow(rowB))
+  }
+
+  if (Cell != undefined) {
+    commonDef.cell =
+      typeof Cell === 'function'
+        ? (props) => Cell({ column: prepareColumn(props.column), row: prepareRow(props.row), value: props.getValue() })
+        : Cell
+  }
+
+  if (filterable !== undefined) {
+    commonDef.enableColumnFilter = filterable
+  }
+
+  if (disableSortBy !== undefined) {
+    commonDef.enableSorting = !disableSortBy
+  }
+
+  if (accessor != undefined) {
+    if (typeof accessor === 'function') {
+      return {
+        id: '',
+        ...commonDef,
+        accessorFn: (value) => accessor(value),
+      }
+    }
+
+    return {
+      ...commonDef,
+      accessorKey: accessor,
+    }
+  }
+
+  return commonDef as ColumnDef<D>
+}
+
+const getColumnVisibility = (hiddenColumns?: string[]) => {
+  const result: Record<string, boolean> = {}
+
+  hiddenColumns?.forEach((column) => {
+    result[column] = false
+  })
+
+  return result
+}
+
+export const getTableInitialState = ({
+  hiddenColumns,
+  columnOrder,
+  sortBy,
+  columnResizing,
+  paginationOptions: pagination,
+}: InitialState): InitialTableState => ({
+  pagination,
+  columnOrder,
+  columnSizing: columnResizing?.columnWidths ?? {},
+  columnVisibility: getColumnVisibility(hiddenColumns),
+  sorting: sortBy?.map(({ id, desc }) => ({ id, desc: !!desc })) ?? [],
+})
+
+export const getCellStyle = ({ size, minSize, maxSize }: { size: number; minSize?: number; maxSize?: number }): CSSProperties => {
+  // fix unit tests console error "`NaN` is an invalid value"
+  const width = isNaN(size) ? 0 : size
```

Comment:
Modifying the program code to fix tests is never a good solution, but if it requires too much time to find the real problem with tests, I think this check won't harm anyone.

Reply:
fixed